recursive-install.js

```diff
@@ -5,58 +5,101 @@ var shell = require('shelljs')
+function noop(x) {
+    return x
 }
 
-function npmInstall (dir) {
-  var exitCode = 0;
-  try {
-    if(argv.production) {
-      console.log('Installing ' + dir + '/package.json with --production option')
-      execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', { cwd: dir})
-      execSync('npm install --production', { cwd: dir})
-    } else {
-      console.log('Installing ' + dir + '/package.json')
-      execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', { cwd: dir})
-      execSync('npm install', { cwd: dir})
+function getPackageJsonLocations(dirname) {
+    return shell.find(dirname)
+        .filter(function (fname) {
+            return !(fname.indexOf('node_modules') > -1 || fname[0] === '.') &&
+                path.basename(fname) === 'package.json'
+        })
+        .map(function (fname) {
+            return path.dirname(fname)
+        })
+}
+
+function npmInstall(dir) {
+    var exitCode = 0;
+    try {
+
+        /**
+         * Environment variable called "EXCLUDE_FOLDERS" with an array of strings. Note: case-sensitive
+         * Ex: packages, containers, etc..
+         */
+        const excludeFolders = process.env.EXCLUDE_FOLDERS;
+
+        if (excludeFolders && typeof excludeFolders === 'string') {
+            const foldersToExclude = excludeFolders
+                .split(",")
+                .map((str) => {
+                    /**
+                     * Remove white space from the beginning or end of a string
+                     */
+                    if (typeof str === 'string') return str.trim();
+                });
+
+            if(foldersToExclude && Array.isArray(foldersToExclude) && foldersToExclude.length > 0) {
+                /**
+                 * Check if current folder needs to be excluded
+                 */
+                let foundFolderToExclude = false;
+
+                const currentPathFolders = dir.split("/");
+
+                for(const folder of foldersToExclude) {
+                    foundFolderToExclude = currentPathFolders.find((x) => x === folder);
+                }
+
+                /**
+                 * If folder to exclude found, return without installing packages
+                 */
+                if(foundFolderToExclude) {
+                    return {
+                        dirname: dir,
+                        exitCode: exitCode
+                    }
+                }
+            }
+        }
+
+        if (argv.production) {
+            console.log('Installing ' + dir + '/package.json with --production option')
+            execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', {cwd: dir})
+            execSync('npm install --production', {cwd: dir})
+        } else {
+            console.log('Installing ' + dir + '/package.json')
+            execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', {cwd: dir})
+            execSync('npm install', {cwd: dir})
+        }
```

Comment:
seems you could use `argv && argv.production === true ? "--production" : ""` to avoid some code duplication